File: node/test/problem_03_test.js
var prb = require("../problems_01_25/problem_03.js");
var assert = require("assert");

describe("Problem_03", function() {
    describe("#LargestPrimeFactor()", function() {
        // the largest prime factor of 10 is 5
        it("should return 5 when the input is 10", function() {
            assert.equal(5, prb.LargestPrimeFactor(10));
        });

        // the largest prime factor of 13195 is 29
        it("should return 5 when the input is 10", function() {
            assert.equal(29, prb.LargestPrimeFactor(13195));
        });
    });
});

File: python/tests/problem_03_test.py
from problems.problems_01_25.problem_03 import largest_prime_factor
import unittest

class Problem03Test(unittest.TestCase):
    """
    test the implementation of Project Euler Problem 03
    """

    def test_case_where_n_equals_10(self):
        """
        the largest prime factor of 10 is 5
        """
        self.assertEqual(largest_prime_factor(10), 5)

    def test_case_where_n_equals_13195(self):
        """
        the largest prime factor of 13195 is 29
        """
        self.assertEqual(largest_prime_factor(13195),29)
